src/paginate.ts

```diff
@@ -297,25 +314,19 @@ export async function paginate<T extends ObjectLiteral>(
         queryBuilder.andWhere(
             new Brackets((qb: SelectQueryBuilder<T>) => {
                 for (const column of searchBy) {
-                    const propertyPath = (column as string).split('.')
-                    const hasRelation =
-                        propertyPath.length > 1 &&
-                        queryBuilder.expressionMap.mainAlias.metadata.hasRelationWithPropertyPath(propertyPath[0])
-
-                    if (['postgres', 'cockroachdb'].includes(queryBuilder.connection.options.type)) {
-                        const alias = hasRelation ? `"${qb.alias}"_` : `"${qb.alias}".`
-                        let columns = ''
-
-                        for (const property of propertyPath) {
-                            columns += `"${property}".`
-                        }
-                        const aliasColumn = alias + columns.substring(0, columns.length - 1)
-
-                        qb.orWhere(`${aliasColumn}::text ILIKE(:search)`, { search: `%${query.search}%` })
-                    } else {
-                        const aliasColumn = hasRelation ? `${qb.alias}_${column}` : `${qb.alias}.${column}`
-                        qb.orWhere(`UPPER(${aliasColumn}) LIKE UPPER(:search)`, { search: `%${query.search}%` })
+                    const property = getPropertiesByColumnName(column)
+                    const { isVirtualProperty, query: virtualQuery } = extractVirtualProperty(qb, property)
+                    const isRelation = checkIsRelation(qb, property.propertyPath)
+
+                    const alias = fixColumnAlias(property, qb.alias, isRelation, isVirtualProperty, virtualQuery)
+                    const condition: WherePredicateOperator = {
+                        operator: 'ilike',
+                        parameters: [alias, `:${column}`],
                     }
+
+                    qb.orWhere(qb['createWhereConditionExpression'](condition), {
```

Comment:
just checkin this line here:

https://github.com/typeorm/typeorm/blob/defb409f5650fed0b7a4ff2933208282a45572fb/src/query-builder/QueryBuilder.ts#L1050

seems like we can avoid regression by adding ::text cast to `condition.parameters[0]` for `postgres` and `cockroachdb`.

```typescript
if (
    driver.options.type === "postgres" ||
    driver.options.type === "cockroachdb"
) {
    condition.parameters[0] += '::text'
}
```

wdyt?

sad we don't have coverage to test for those drivers..

Reply:
yes without a test we cannot check it deep. But for now and without any additional test by the involved parts lgtm.